code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/HealthCheckReconcile.java

```diff
@@ -0,0 +1,39 @@
+package io.cattle.iaas.healthcheck.process;
+
+import io.cattle.iaas.healthcheck.service.HealthcheckService;
+import io.cattle.platform.core.constants.HealthcheckConstants;
+import io.cattle.platform.core.model.HealthcheckInstanceHostMap;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPostListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessLogic;
+import io.cattle.platform.util.type.Priority;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named
+public class HealthCheckReconcile extends AbstractObjectProcessLogic implements ProcessPostListener, Priority {
+
+    @Inject
+    HealthcheckService hcSvc;
+
+    @Override
+    public String[] getProcessNames() {
+        return new String[] { "healthcheckinstancehostmap.remove" };
+    }
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        HealthcheckInstanceHostMap hcihm = (HealthcheckInstanceHostMap) state.getResource();
+        hcSvc.healthCheckReconcile(hcihm, HealthcheckConstants.HEALTH_STATE_UNHEALTHY);
```

Comment:
@cjellick when the host gets removed/disconnected. So when it happens, I set the reported state as unhealthy. Then in underlying code, we check what non-removed hosts are saying. If they all say unhealthy - then host is considered as unhealthy. If one of them says healthy - host is considered as healthy.

I've picked unhealthy event over healthy because a single "healthy" event brings host to healthy even if the remaining hosts say its unhealthy. While "unhealthy" doesn't have this power.